src/VisualStudio/Core/Test.Next/Services/LanguageServiceTests.cs

```diff
@@ -0,0 +1,138 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Editor.UnitTests.Workspaces;
+using Microsoft.CodeAnalysis.Remote;
+using Microsoft.CodeAnalysis.Test.Utilities;
+using Microsoft.VisualStudio.LanguageServer.Protocol;
+using Newtonsoft.Json.Linq;
+using Roslyn.Test.Utilities.Remote;
+using StreamJsonRpc;
+using Xunit;
+
+namespace Roslyn.VisualStudio.Next.UnitTests.Services
+{
+    [UseExportProvider]
+    public class LanguageServiceTests
+    {
+        [Fact, Trait(Traits.Feature, Traits.Features.RemoteHost)]
+        public async Task CSharpLanguageServiceTest()
```

Comment:
please reuse the exacting set of nav2 tests that we have.  We want to be able to understand how all those scenarios work here and if/how they are broken.  writing a new set of tests means we have no insight on how the work to make LSP rich enogh to support this scenario are.

This is equivalent to what we did with completion when moving to async completion.  we didn't write new async compleiton tests.  We ensured that we were running all the old tests and thus could know about all regressions.  We need that here as well.

Reply:
no reason to re-run the same tests. this just makes sure LSP side get connected and do sanity check. existing tests cover actual services already.

Reply:
There is a reason.  We want to ensure that we understand regressions and that none of the well-tested existing codepaths have any problem being hit in LSP.

This is something we did with both far and navto when we moved OOP.  We didn't just test a tiny handful of cases.  We tested *100%* of cases. And it *did* find issues.  You're now accessing the entire navto stack through a new entrypoint.  we should ensure the entire navto stacks works 100% ok through that entrypoint.

Reply:
For example:

```c#
        protected async Task TestAsync(string content, Func<TestWorkspace, Task> body)
        {
            // Keep track of tested combinations to ensure all expected tests run
            var testedCombinations = new HashSet<(bool outOfProcess, Type documentTrackingServiceType)>();

            await TestAsync(content, BodyWrapper, outOfProcess: true);
            await TestAsync(content, BodyWrapper, outOfProcess: false);
```

Doing this caught a *lot* of issues.  Same with when we did it for FindRefs.  Same when we did it for async completion.   We also did it when you added "semantic span analysis".  We moved a ton of IDE features over to hammering it by running their same scenario with normal semantic-models, and with your "lightweight semantic span model".

It's often the case that something will try to use a complex subsystem in a new way, and it may end up causing something unintentional to happen.  Being able to just hammer at it with hundreds to thousands of real tests ensures that we don't have corners of the code that aren't tested and which may have issues we didn't anticipate.